docs/tutorials/mongoid-configuration.txt

```diff
@@ -30,52 +30,86 @@ i.e. ``development``, ``test`` or ``production``. The third level key,
 ``default`` in the above example, refers to the Mongo client name.
 Most applications will use a single client named ``default``.
 
-Rails Applications
-******************
+Generating Default Configuration
+********************************
 
-If you are using Rails, you can have Mongoid generate a default configuration
-file for you by running the following command:
+If you are using Ruby on Rails, you can have Mongoid generate a default
+configuration file for you by running the following command:
 
 .. code-block:: bash
 
-  $ rails g mongoid:config
+  rails g mongoid:config
 
 The configuration file will be placed in ``config/mongoid.yml``.
 
-Environment Determination
-*************************
+If you are not using Ruby on Rails, you can copy the minimal configuration
+given above and save it as ``config/mongoid.yml``.
+
+Loading Mongoid Configuration
+*****************************
+
+If you are using Ruby on Rails, Mongoid configuration is automatically loaded
+for the current environment as stored in ``Rails.env`` when the application
+loads.
 
-When Mongoid loads its configuration, it chooses the environment to use
-in the following order:
+You may need to configure the ORM for your application to be Mongoid by
+adding the following to ``application.rb``:
+
+.. code-block:: ruby
 
-- ``Rails.env`` if using Rails.
-- ``Sinatra::Base.environment`` if using Sinatra.
-- ``RACK_ENV`` environment variable.
-- ``MONGOID_ENV`` environment variable.
+  config.generators do |g|
+    g.orm :mongoid
+  end
 
-If you are not building a Rack based application and want to override the
-environment programatically, you can pass a second parameter to ``load!``
-as follows:
+If you are not using Ruby on Rails, Mongoid configuration must be loaded
+manually. This can be done via the ``Mongoid.load!`` method, which takes
+the configuration file path as its argument, as follows:
 
 .. code-block:: ruby
 
+  # Use automatically detected environment name
+  Mongoid.load!("path/to/your/mongoid.yml")
+  
+  # Specify environment name manually
   Mongoid.load!("path/to/your/mongoid.yml", :production)
```

Comment:
Where do you put this code block? Also in `application.rb`? Or does it depend?

Reply:
This section applies to non-Rails applications. As such the bootstrap code can vary from one application to another.

Example for tenex: https://github.com/p-mongo/tenex/blob/master/lib/fe/boot.rb#L5

Reply:
That makes sense. Thanks for clarifying.